app/models/cmr.rb

```diff
@@ -355,24 +355,45 @@ def self.get_raw_granule(concept_id)
     granule_results["Granule"]
   end
 
+  # Given a granule concept id, it will pull all revisions and return the latest granule revision.
   def self.get_raw_granule_results(concept_id)
+    url = Cmr.api_url("granules", "umm_json", {"concept_id" => concept_id})
+    granule_json_str = Cmr.cmr_request(url).parsed_response
+    granule_dict = JSON.parse(granule_json_str)
+    if granule_dict['hits'].to_i == 0
+      raise CmrError, "CMR returned 0 hits for #{concept_id}"
+    end
+    # Access to first element because it should have only one record (concept_id)
+    raw_format = granule_dict['items'][0]['meta']['format']
+    format_type = raw_format == 'application/vnd.nasa.cmr.umm+json' ? 'umm_json' : 'echo10'
+
+    if format_type == 'echo10'
+      granule_result_echo10 = get_raw_granule_results_echo10(concept_id)
+      granule_result_echo10['format_type'] = 'echo10'
+      return granule_result_echo10
```

Comment:
Why does this case return a "granule_result_echo10" but the normal return is a granule_dict?